lib/pure/stats.nim

```diff
@@ -53,6 +53,37 @@ runnableExamples:
   doAssert statistics.kurtosis() ~= -1.0
   doAssert statistics.kurtosisS() ~= -0.7000000000000008
 
+  var 
+    ASv = [7,4,6,3,9,1]        #  sorted   1 3 4 6 7 9
+    A   : array[len(ASv),int]
+
+  func myCmp(x,y:int):int =
+    if x==y : 0 elif x<y : -1 else: 1
+
+  A= ASv;  doAssert quickSelect(A, 2, 5, 2, myCmp) == 1  # smallest element
+  A= ASv;  doAssert quickSelect(A, 2, 5, 5, myCmp) == 9  # largest element
+  #
+  A= ASv;  doAssert median(A,0,5,myCmp)      == 5.0
+  A= ASv;  doAssert median_low(A,0,5,myCmp)  == 4
+  A= ASv;  doAssert median_high(A,0,5,myCmp) == 6
+  #
+  A= ASv;  doAssert median(A,2,5)            == 4.5
+  A= ASv;  doAssert median_low(A,2,5)        == 3
+  A= ASv;  doAssert median_high(A,2,5)       == 6
+  #
+  A= ASv;  doAssert median(A,0,4,myCmp)      == 6.0
+  A= ASv;  doAssert median_low(A,0,4,myCmp)  == 6
+  A= ASv;  doAssert median_high(A,0,4,myCmp) == 6
```

Comment:
This is too much for the module level `runnableExamples` imo, it should go to the documentation of the repsective functions or tests.

Reply:
Done